### ISSUE TITLE
Reflections is not binary compatible with Google Guava 11.0-rc1

### ISSUE THREAD

**Issue by GoogleCodeExporter** [NickstaDB/reflections#94]
```
I am aware that Guava 11 has not been released yet, but I guess it makes sense 
to give an early heads up. In 11.0-rc1 (and presumably also the final release) 
CacheBuilder#maximumSize(int) has been replaced with 
CacheBuilder#maximumSize(long). This causes a runtime issue:

java.lang.NoSuchMethodError: 
com.google.common.cache.CacheBuilder.maximumSize(I)Lcom/google/common/cache/Cach
eBuilder;
        at org.reflections.adapters.JavassistAdapter.<init>(JavassistAdapter.java:28)
        at org.reflections.util.ConfigurationBuilder.<init>(ConfigurationBuilder.java:41)
        ...

Cheers,
Stephan
```

Original issue reported on code.google.com by `stephan...@gmail.com` on 16 Dec 2011 at 2:00

**Comment by GoogleCodeExporter** [comment]
```
For reference, in the mean time Guava 11.0 has been released.
```

Original comment by `stephan...@gmail.com` on 30 Dec 2011 at 12:29

**Comment by GoogleCodeExporter** [comment]
```
I too am running into this issue with Guava 11.0
```

Original comment by `ch...@edatasource.com` on 9 Jan 2012 at 10:09

**Comment by GoogleCodeExporter** [comment]
```
Same here...
```

Original comment by `eide...@gmail.com` on 12 Jan 2012 at 7:14

**Comment by GoogleCodeExporter** [comment]
```
I checked out the code and tried to bump the Guava dependency version. Turns 
out that one of the unit tests relies on the now-removed 
Files#deleteRecursively method. This call can be replaced with 
FileUtils#deleteDirectory, at the cost of introducing commons-io as an 
additional dependency with scope "test". I'd say that's a very minor issue.

I have attached a patch for those interested.
```

Original comment by `stephan...@gmail.com` on 19 Jan 2012 at 10:43

Attachments:
- [reflections.issue94.patch](https://storage.googleapis.com/google-code-attachments/reflections/issue-94/comment-4/reflections.issue94.patch)

**Comment by GoogleCodeExporter** [comment]
```
Using revision 150 (0.9.7-SNAPSHOT, fix some issues), with the following 
dependency:

<dependency>
    <groupId>com.google.guava</groupId>
    <artifactId>guava</artifactId>
    <version>11.0.1</version>
</dependency>

Unfortunately there is a compile error in the org.reflections.VfsTest.java 
file.  To get around this, in the VfsTest.java file I replaced 
"Utils.deleteRecursively(dir1);" with "Utils.deleteRecursively(dir1);" and I 
added the following method to org.reflections.util.Utils:

    public static void deleteRecursively( File file ) throws IOException
    {
        if ( file.isDirectory() )
        {
            for ( File child : file.listFiles() )
            {
                deleteRecursively( child );
            }
        }
        if ( !file.delete() )
        {
            throw new FileNotFoundException( "Failed to delete file: " + file ); //$NON-NLS-1$
        }
    }

* While I could use the commons-io FileUtils#deleteDirectory method, I would 
prefer one less dependency.
```

Original comment by `mark.sor...@gmail.com` on 6 Mar 2012 at 9:10

**Comment by GoogleCodeExporter** [comment]
```
Sorry I meant that the Guava deprecated and removed method "deleteRecursively", 
be replaced with a new method consisting of the sample code above.
```

Original comment by `mark.sor...@gmail.com` on 8 Mar 2012 at 6:14

**Comment by GoogleCodeExporter** [comment]
```
I'll not bother to create another issue ticket, but suffice it to say that an 
upgrade to Guava 12.0-rc1 introduces new incompatibilities. See the attached 
patch for the (trivial) fix.
```

Original comment by `stephan...@gmail.com` on 31 Mar 2012 at 9:58

Attachments:
- [guava-12.0-rc1.patch](https://storage.googleapis.com/google-code-attachments/reflections/issue-94/comment-7/guava-12.0-rc1.patch)

**Comment by GoogleCodeExporter** [comment]
```
Same problem here, I cannot use reflections-0.9.6 with guava-11.0.

Thanks.
```

Original comment by `espina.e...@gmail.com` on 12 Apr 2012 at 12:43

**Comment by GoogleCodeExporter** [comment]
```
Is there any news on when a fix will be available for using 0.9.6 with 
guava-11.0+? 
```

Original comment by `richardd...@gmail.com` on 18 Apr 2012 at 4:44

**Comment by GoogleCodeExporter** [comment]
```
guys. this is serious.  can we get a release out?
```

Original comment by `adrian.f...@gmail.com` on 25 Apr 2012 at 5:14

**Comment by GoogleCodeExporter** [comment]
```
> guys. this is serious.  can we get a release out?

+1. A release would be very helpful.
```

Original comment by `sharedo...@gmail.com` on 25 Apr 2012 at 6:55

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
A new release will be really useful
```

Original comment by `andrea.t...@gmail.com` on 26 Apr 2012 at 6:56

**Comment by GoogleCodeExporter** [comment]
```
For impatients,  there is a temporary solution to make reflections work with 
guava 12.0-rc2

Clone this mirror repo https://github.com/hadashi/reflections.git, which is a 
clone of reflections svn repo, where patch is applied then run
mvn install -DskipTests=true

```

Original comment by `andrea.t...@gmail.com` on 27 Apr 2012 at 5:03

**Comment by GoogleCodeExporter** [comment]
```
Issue 107 has been merged into this issue.
```

Original comment by `ronm...@gmail.com` on 2 May 2012 at 3:44

**Comment by GoogleCodeExporter** [comment]
```
fixed on 0.9.7-RC1 and trunk. comments?
```

Original comment by `ronm...@gmail.com` on 2 May 2012 at 3:52
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Great! I can confirm that this issue is fixed.

I see that r151 bumps the Guava dependency to 12.0-rc1. Note that 12.0 was 
released a few days ago :)
```

Original comment by `stephan...@gmail.com` on 2 May 2012 at 4:05

**Comment by GoogleCodeExporter** [comment]
```
Thanks guys, I can confirm even for me it's working fine.
```

Original comment by `andrea.t...@gmail.com` on 2 May 2012 at 4:07

**Comment by GoogleCodeExporter** [comment]
```
works for me, too
```

Original comment by `adrian.c...@cloudsoftcorp.com` on 2 May 2012 at 10:51

**Comment by GoogleCodeExporter** [comment]
```
fixed 0.9.7
```

Original comment by `ronm...@gmail.com` on 22 May 2012 at 7:47
- Changed state: **Fixed**
